### PR TITLE
soplex 6.0.3 (new formula)

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Related to https://github.com/Homebrew/homebrew-core/pull/118774 since `soplex` is part of the scip-optsuite.

### CODE REVIEW

File: Formula/soplex.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "gmp"
+  depends_on "tbb"
+  uses_from_macos "zlib"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    cd "build" do
+      system "make"
+      system "make", "install"
+    end
```
Comment (chenrui333) [inline Formula/soplex.rb:19]: maybe like this?
```suggestion
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

File: Formula/soplex.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      Bounds
+       0 <= x1 <= 40
+       2 <= x4 <= 3
+      General
+       x4
+      End
+    EOS
+    assert_match "problem is solved [optimal]", shell_output("#{bin}/soplex test.lp")
+
+    system ENV.cxx, pkgshare/"example.cpp", "-std=c++14", "-L#{lib}", "-I#{include}",
```
Comment (chenrui333) [inline Formula/soplex.rb:40]: can we use c++17?

Comment (carlocab) [reply Formula/soplex.rb:40]: The project uses C++14, so we should use that.

Comment (chenrui333) [reply Formula/soplex.rb:40]: yeah, make sense!

**Review by chenrui333 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]
I'll open a follow-up PR to add a `livecheck` block.

**Pull request closed without merging**